api/s2n.h

```diff
@@ -3931,6 +3931,79 @@ S2N_API int s2n_connection_deserialize(struct s2n_connection *conn, uint8_t *buf
  */
 S2N_API int s2n_config_set_cert_authorities_from_trust_store(struct s2n_config *config);
 
+/* Append the distinguished name for a certificate authority to the config.
+ *
+ * The full set of certificate authorities will be advertised during the handshake.
+ * This can help your peer select a certificate if they have multiple certificate
+ * chains available.
+ *
+ * For now, s2n-tls only supports advertising certificate authorities to support
+ * client auth, so only servers will send the list of certificate authorities.
+ *
+ * s2n-tls currently limits the total certificate authorities size to 10k bytes.
+ * This method will fail if the total length of the certificate authorities
+ * configured exceeds that limit.
+ *
+ * @param config A pointer to the s2n_config object.
+ * @returns S2N_SUCCESS on success. S2N_FAILURE on failure.
+ */
+S2N_API int s2n_config_add_cert_authority(struct s2n_config *config, uint8_t *name, uint16_t length);
+
+struct s2n_certificate_authority_list;
+
+/**
+ * Checks whether the offered list has a next entry.
+ *
+ * @param list A pointer to the certificate authority list being read.
+ * @returns bool If true, there is a next entry.
+ */
+S2N_API bool s2n_certificate_authority_list_has_next(struct s2n_certificate_authority_list *list);
+
+/**
+ * Obtains the next entry in the certificate authority list.
+ *
+ * The lifetime of the name is bound to the lifetime of the list (in other
+ * words, it must not be used past the certificate request callback).
+ *
+ * @param list A pointer to the list being read.
+ * @param name An out-pointer to the next name in the list, will be null if there's no next name.
```

Comment:
We'll probably want to think more about this API -- today the `name` here is a DER-encoded value for the DistinguishedName, which is potentially somewhat annoying for consumers of this API.